File: Flutter/pickup_scheduler/lib/utils/BoundingBox.dart
import 'dart:math';

import 'package:pickup_scheduler/utils/Customer.dart';

class BoundingBox {
  double minX;
  double maxX;
  double minY;
  double maxY;

  BoundingBox(List<Customer> customers) {
    minX = customers[0].x;
    maxX = customers[0].x;
    minY = customers[0].y;
    maxY = customers[0].y;

    for (Customer c in customers) {
      minX = (c.x < minX) ? c.x : minX;
      maxX = (c.x > maxX) ? c.x : maxX;
      minY = (c.y < minX) ? c.y : minY;
      maxY = (c.y > maxY) ? c.y : maxY;
    }
  }

  Point getRandomPointInBounds() {
    var rng = new Random();
    //Interpolate randomly between min and max
    double x = ((rng.nextDouble() * (maxX - minX)) + minX);
    double y = ((rng.nextDouble() * (maxY - minY)) + minY);
    return new Point(x, y);
  }
}

File: Native Android/app/src/main/java/com/ibisrecycling/lars/pickupscheduler/utils/BoundingBox.kt
package com.ibisrecycling.lars.pickupscheduler.utils

class BoundingBox(customers: Array<Customer>) {
	private var minX: Double
	private var maxX: Double
	private var minY: Double
	private var maxY: Double

	init {
		minX = customers[0].x
		maxX = customers[0].x
		minY = customers[0].y
		maxY = customers[0].y

		for (c in customers) {
			minX = if (c.x < minX) c.y else minX
			maxX = if (c.x > maxX) c.y else maxX
			minY = if (c.x < minY) c.y else minY
			maxY = if (c.x > maxY) c.y else maxY
		}
	}

	private fun Double.isOutOfBoundsX(): Boolean = this > maxX || this < minX
	private fun Double.isOutOfBoundsY(): Boolean = this > maxY || this < minY

	fun getRandomPointInBounds(): Point {
		var x: Double
		var y: Double
		do {
			x = Math.random()
		} while (x.isOutOfBoundsX())
		do {
			y = Math.random()
		} while (y.isOutOfBoundsY())
		return Point(x, y)
	}
}